src/olympia/addons/tests/test_commands.py

```diff
@@ -479,3 +480,80 @@ def test_correctly_ignored_because_strict_compatibility_is_enabled(self):
+        addon_factory(version_kw={'application': amo.ANDROID.id})
+        addon_with_both_firefoxes = addon_factory()
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id,
+            version=addon_with_both_firefoxes.current_version,
+            min=av_min, max=av_max)
+        addon_with_thunderbird_and_android = addon_factory(
+            version_kw={'application': amo.THUNDERBIRD.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id,
+            version=addon_with_thunderbird_and_android.current_version,
+            min=av_min, max=av_max)
+        addon_with_firefox_and_seamonkey = addon_factory(
+            version_kw={'application': amo.FIREFOX.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.SEAMONKEY.id,
+            version=addon_with_firefox_and_seamonkey.current_version,
+            min=av_seamonkey_min, max=av_seamonkey_max)
+
+        # Those add-ons should be deleted as they are only compatible with
+        # Thunderbird or Seamonkey, or both.
+        addon = addon_factory(version_kw={'application': amo.THUNDERBIRD.id})
+        addon2 = addon_factory(version_kw={'application': amo.SEAMONKEY.id,
+                                           'min_app_version': '2.49.3',
+                                           'max_app_version': '2.49.*'})
+        addon3 = addon_factory(version_kw={'application': amo.THUNDERBIRD.id})
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.SEAMONKEY.id,
+            version=addon3.current_version,
+            min=av_seamonkey_min, max=av_seamonkey_max)
+
+        # We've manually changed the ApplicationVersions, so let's run
+        # update_appsupport(). In the real world tthat is done automatically
+        # when uploading a new version, either through the version_changed
+        # signal or via a cron that catches any versions that somehow fell
+        # through the cracks once a day.
+        update_appsupport(Addon.objects.values_list('pk', flat=True))
+
+        assert Addon.objects.count() == 8
+
+        with count_subtask_calls(
+                pa.delete_addon_not_compatible_with_firefoxes) as calls:
+            self.make_the_call()
+
+        assert len(calls) == 1
+        assert calls[0]['kwargs']['args'] == [[addon.pk, addon2.pk, addon3.pk]]
+        assert not Addon.objects.filter(pk=addon.pk).exists()
+        assert not Addon.objects.filter(pk=addon2.pk).exists()
+        assert not Addon.objects.filter(pk=addon3.pk).exists()
+
+        assert Addon.objects.count() == 5
+
+        # Make sure ApplicationsVersions targeting Thunderbird or Seamonkey are
+        # gone.
+        assert not ApplicationsVersions.objects.filter(
+            application=amo.SEAMONKEY.id).exists()
+        assert not ApplicationsVersions.objects.filter(
+            application=amo.THUNDERBIRD.id).exists()
```

Comment:
Check `AppSupport` is cleared/regenerated too.

Reply:
36226b917409ab0378e5f953a0841ec01f7ac319